### PR TITLE
refactor(package): cross-platform `prepare` script (`scripts`)

### LINKED ISSUE #1497: `npm install` fails on Windows
* Operating System: Windows
* Node Version: 8.9.4
* NPM Version: 6.4.0
* webpack Version: N/A
* webpack-dev-server Version: 3.1.8

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code
N/A

### Expected Behavior
`npm install` finishes without error

### Actual Behavior
`npm install` finishes with the following error because `rm` is not present at Windows environments

```bash
c:\path>git clone https://github.com/webpack/webpack-dev-server.git
Cloning into 'webpack-dev-server'...
remote: Counting objects: 4389, done.
remote: Compressing objects: 100% (25/25), done.
remote: Total 4389 (delta 12), reused 14 (delta 5), pack-reused 4359
Receiving objects: 100% (4389/4389), 1.99 MiB | 2.02 MiB/s, done.
Resolving deltas: 100% (2432/2432), done.

c:\path>cd webpack-dev-server

c:\path\webpack-dev-server>npm install

> webpack-dev-server@3.1.8 prepare c:\path\webpack-dev-server
> (rm ssl/*.pem || true) && npm run -s transpile:index && npm run -s build:live && npm run -s build:index && npm run -s build:sockjs

'rm' is not recognized as an internal or external command,
operable program or batch file.
'true' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! webpack-dev-server@3.1.8 prepare: `(rm ssl/*.pem || true) && npm run -s transpile:index && npm run -s build:live && npm run -s build:index && npm run -s build:sockjs`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the webpack-dev-server@3.1.8 prepare script.
```

### For Bugs; How can we reproduce the behavior?
On an Windows environment, run the following commands

```bash
git clone https://github.com/webpack/webpack-dev-server.git
cd webpack-dev-server
npm install
```

### PR DESCRIPTION
**Pull request by ferdinando-ferreira**
Given that it is a simple removal there was no need to import a new dependency (`rimraf`, for instance)

### Type 

- [x] Refactor

### Issues

- Fixes #1497

### SemVer

- [x] Patch

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   "scripts": {
     "lint": "eslint bin lib test examples client-src",
     "test": "nyc --reporter lcovonly mocha --full-trace --check-leaks --exit",
-    "prepare": "(rm ssl/*.pem || true) && npm run -s transpile:index && npm run -s build:live && npm run -s build:index && npm run -s build:sockjs",
+    "prepare": "./node_modules/.bin/rimraf ./ssl/*.pem && npm run -s transpile:index && npm run -s build:live && npm run -s build:index && npm run -s build:sockjs",
```
Comment (michael-ciniawsky) [inline package.json]:
```
"prepare": "rimraf ./ssl/*.pem && ..."
```

`npm` should look up `node_modules/.bin` automatically or didn't this work for you (on windows)?

**Comment by ferdinando-ferreira** [comment]
> Please use e.g `del-cli` or rimraf instead and remove this file

```diff
"prepare: "del-cli ssl/*.pem && ..."
- "prepare:delete-certs: ..."
```
@michael-ciniawsky: done and done. Although I have no objection to the requested changes (and implemented as requested) the original decision to create a custom delete with a standalone rule was
* to avoid pulling a dependency for something as trivial as removing some files from a folder
* to allow for the certificates to be removed from the folder even when prepare is not called directly.

File: package.json (hunk 1 above)
Comment (ferdinando-ferreira) [reply package.json]: On Windows it doesn't automatically look into that folder

```console
c:\path\webpack-dev-server>.\node_modules\.bin\rimraf ssl/*.pem

c:\path\webpack-dev-server>rimraf ssl/*.pem
'rimraf' is not recognized as an internal or external command,
operable program or batch file.
```

File: package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "lint": "eslint bin lib test examples client-src",
     "test": "nyc --reporter lcovonly mocha --full-trace --check-leaks --exit",
     "prepare": "npm run -s prepare:delete-certs && npm run -s transpile:index && npm run -s build:live && npm run -s build:index && npm run -s build:sockjs",
-    "prepare:delete-certs": "node lib/delete-certs.js",
+    "prepare:delete-certs": "./node_modules/.bin/rimraf ./ssl/*.pem",
```
Comment (evilebottnawi) [inline package.json]: you can avoid `./node_modules/.bin/`

Comment (ferdinando-ferreira) [reply package.json]: @michael-ciniawsky beat you to the punch. Already fixed :)

**Comment by michael-ciniawsky** [comment]
> to avoid pulling a dependency for something as trivial as removing some files from a folder

While e.g `rimraf` may contain slightly more code, it fits the use case (cross-platform `rm`), is tested and clearly separates itself from the rest of the code base (e.g `delete-certs.js` wouldn't have belonged into `lib`, since it's a build helper). Size doesn't really matter here since this module doesn't run in e.g the browser

> to allow for the certificates to be removed from the folder even when prepare is not called directly.

You could propose a `clean` script and refactored it out of `prepare`, **if** it has a concrete benefit/use case. But please open a separate PR for this

**Comment by michael-ciniawsky** [comment]
@ferdinando-ferreira Please sign the CLA

**Comment by ferdinando-ferreira** [comment]
@michael-ciniawsky: 
> licence/cla Expected — Waiting for status to be reported 
I believe I have to wait until the cla system somehow update it's status. Maybe something related to https://github.com/cla-assistant/cla-assistant/issues/124?

**Comment by ferdinando-ferreira** [comment]
@michael-ciniawsky:  licence/cla stuck at
> Expected — Waiting for status to be reported 

May be related to https://github.com/cla-assistant/cla-assistant/issues/303 and cla-assistant/cla-assistant#124, should I do something?

**Pull request merged**